qsc/plot.py

```diff
@@ -0,0 +1,165 @@
+        else:
+            r2 = 0
+        return r0+r*r1+r**2*r2
+    def zSurf(r,phi,theta):
+        thetaN = theta-(self.iota-self.iotaN)*phi
+        z0 = Zaxisf(phi)
+        z1 = (          X1cF(phi)*np.cos(thetaN)  +X1sF(phi)*np.sin(thetaN)  )*normalZ(phi)+(          Y1cF(phi)*np.cos(thetaN)  +Y1sF(phi)*np.sin(thetaN)  )*binormalZ(phi)
+        if self.order!='r1':
+            z2 = (X20F(phi)+X2cF(phi)*np.cos(2*thetaN)+X2sF(phi)*np.sin(2*thetaN))*normalZ(phi)+(Y20F(phi)+Y2cF(phi)*np.cos(2*thetaN)+Y2sF(phi)*np.sin(2*thetaN))*binormalZ(phi)+(Z20F(phi)+Z2cF(phi)*np.cos(2*thetaN)+Z2sF(phi)*np.sin(2*thetaN))*tangentZ(phi)
+        else:
+            z2 = 0
+        return z0+r*z1+r**2*z2
+
+    # Create one plot with several cuts at different toroidal planes
+    fig = plt.figure(figsize=(6, 6), dpi=80)
+    ax  = plt.gca()
+    theta = np.linspace(0,2*np.pi,ntheta)
+    phi1D = np.linspace(0,2*np.pi/self.nfp,nphiRZ)
+    for phi in phi1D:
+        rSurfi=rSurf(rPlot,phi,theta)
+        zSurfi=zSurf(rPlot,phi,theta)
+        if phi*self.nfp/(2*np.pi)==0:
+            label = r'$\phi$=0'
+        elif phi*self.nfp/(2*np.pi)==0.25:
+            label = r'$\phi={\pi}/$'+str(2*self.nfp)
+        elif phi*self.nfp/(2*np.pi)==0.5:
+            label = r'$\phi=\pi/$'+str(self.nfp)
+        elif phi*self.nfp/(2*np.pi)==0.75:
+            label = r'$\phi={3\pi}/$'+str(2*self.nfp)
+        elif phi*self.nfp/(2*np.pi)==1.0:
+            continue
+        else:
+            label = '_nolegend_'
+        color = next(ax._get_lines.prop_cycler)['color']
+        plt.plot(rSurfi,zSurfi,linewidth=2,label=label,color=color)
+        plt.plot(Raxisf(phi),Zaxisf(phi),'b+',color=color)
+    plt.xlabel('R (meters)')
+    plt.ylabel('Z')
+    plt.legend()
+    plt.tight_layout()
+    ax.set_aspect('equal')
+    if saveFile!=[]:
+        fig.savefig(saveFile+'_poloidal.pdf')
+
+    # Create 3D plot
+    fig = plt.figure()
+    ax  = plt.axes(projection='3d')
+    def Bf(r,phi,theta):
+        if self.order=='r1':
+            return self.B0*(1+r*self.etabar*np.cos(theta-(self.iota-self.iotaN)*phi))
+        else:
+            return self.B0*(1+r*self.etabar*np.cos(theta-(self.iota-self.iotaN)*phi))+r*r*(B20F(phi)+self.B2c*np.cos(theta-(self.iota-self.iotaN)*phi)+self.B2s*np.sin(theta-(self.iota-self.iotaN)*phi))
+    theta1D = np.linspace(0,2*np.pi,ntheta)
+    phi1D   = np.linspace(0,2*np.pi,nphi)
+    Xsurf = np.zeros((ntheta,nphi))
+    Ysurf = np.zeros((ntheta,nphi))
+    Zsurf = np.zeros((ntheta,nphi))
+    Bmag  = np.zeros((ntheta,nphi))
+    for countT,th in enumerate(theta1D):
+        for countP,ph in enumerate(phi1D):
+            rs=rSurf(rPlot,ph,th)
+            zs=zSurf(rPlot,ph,th)
+            Xsurf[countT,countP]=rs*np.cos(ph)
+            Ysurf[countT,countP]=rs*np.sin(ph)
+            Zsurf[countT,countP]=zs
```

Comment:
This logic does not produce the proper surface! You cannot just ignore the phi component of the Frenet vectors. It's probably easiest to evaluate r = r0 + X n + Y b + Z t in Cartesian coordinates.